### PR TITLE
Make sure ios tests fail if they fail

### PR DESCRIPTION
**Pull request by dnfield**
Addresses the fact that the tests aren't failing when they should in https://github.com/flutter/flutter/issues/63742

### CODE REVIEW

**Comment by dnfield** [comment]
Landing this to make the tree red.

File: testing/run_tests.py [hunk 1]
```diff
@@ -370,7 +370,6 @@ def RunObjcTests(ios_variant='ios_debug_sim_unopt'):
   ios_out_dir = os.path.join(out_dir, ios_variant)
   EnsureIosTestsAreBuilt(ios_out_dir)
 
-  pretty = "cat" if subprocess.call(["which", "xcpretty"]) else "xcpretty"
   ios_unit_test_dir = os.path.join(buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests')
 
   command = [
```
Comment (xster) [inline testing/run_tests.py:379]: This removes it locally right? Can we just set -o pipefail && here?

File: testing/ios/IosUnitTests/run_tests.sh [hunk 2]
```diff
@@ -8,9 +8,6 @@ if [ $# -eq 1 ]; then
   FLUTTER_ENGINE=$1
 fi
 
-set -o pipefail && xcodebuild -sdk iphonesimulator \
```
Comment (gaaclarke) [inline testing/ios/IosUnitTests/run_tests.sh:11]: `set -o pipefail` was the magic that worked with xcpretty.  When we removed xcpretty it was no longer magical.

Comment (gaaclarke) [reply testing/ios/IosUnitTests/run_tests.sh:11]: fyi @jmagman

Comment (dnfield) [reply testing/ios/IosUnitTests/run_tests.sh:11]: It wasn't being used in the python script, and I couldn't seem to get it working in a way the script liked. Honestly, I'm not sure its worth it. Every time I've ever had a test failure I had to re-run my tests without xcpretty to figure out why.

(@xster same reply to your comment above)

Comment (gaaclarke) [reply testing/ios/IosUnitTests/run_tests.sh:11]: I'm fine removing xcpretty, just wanted to explain why this was there and how this broke.

File: testing/run_tests.py [hunk 3]
```diff
@@ -370,17 +370,19 @@ def RunObjcTests(ios_variant='ios_debug_sim_unopt'):
   ios_out_dir = os.path.join(out_dir, ios_variant)
   EnsureIosTestsAreBuilt(ios_out_dir)
 
-  pretty = "cat" if subprocess.call(["which", "xcpretty"]) else "xcpretty"
   ios_unit_test_dir = os.path.join(buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests')
 
+  # Avoid using xcpretty unless the following can be addressed:
+  # - Make sure all relevant failure output is printed on a failure.
+  # - Make sure that a failing exit code is set for CI.
+  # See https://github.com/flutter/flutter/issues/63742
```
Comment (jmagman) [inline testing/run_tests.py:378]: Should we even mention this?  I never want to hide the full Xcode build results when there's an error, it's frustrating when I can't reproduce the failure locally.  `xcpretty` is nice for end users, not me who wants to see all the clang flags and search paths.  "relevant failure output" is pretty different depending on what's failing.

Comment (dnfield) [reply testing/run_tests.py:378]: I did that thinking that a year or two from now, someone is going to say "Hey, we should add xcpretty to make this easier to read!" again :)

Comment (jmagman) [reply testing/run_tests.py:378]: And it'll probably be you or me. 😄

**Pull request merged**